salt/states/pkg.py

```diff
@@ -165,6 +165,10 @@ def _fulfills_version_spec(versions, oper, desired_version,
     otherwise returns False
     '''
     cmp_func = __salt__.get('pkg.version_cmp')
+    # stripping "with_origin" dict wrapper
+    if salt.utils.is_freebsd():
+        if type(versions) is dict and 'version' in versions:
```

Comment:
Like I said, we should not be using simple type matching here. Can we get this changed to:

```python
if isinstance(versions, dict) and 'version' in versions:
```